### DISCUSSION TITLE
List ellipsis not working?

### DISCUSSION THREAD

**Discussion by zebulac** [scandum/tintin#117] [Bugs]
I'm trying to extract a portion of a list using the ellipsis and it seems to not be working. Perhaps I'm misunderstanding how it works.

#list temp create;
#list temp add {test0};
#list temp add {test1};
#list temp add {test2};
#list temp add {test3};
#list temp add {test4};
#list temp add {test5};
#list temp add {test6};
#list temp add {test7};
#list temp add {test8};
#list temp add {test9};

#show $temp[2..3] -> {test1}{test2}{test3}{test4}{test5}{test6}{test7}{test8}{test9}
#show $temp[3..] -> {test2}{test3}{test4}{test5}{test6}{test7}{test8}{test9}
#show $temp[-5..] -> {test5}{test6}{test7}{test8}{test9}
#show $temp[-5..-2] -> {test5}{test6}{test7}{test8}{test9}

I was expecting, for example, that $temp[-5..-2] would return {test5}{test6}{test7}{test8}

**Comment by scandum** [comment]
Works for me. Are you using the most recent tintin version?

**Reply by zebulac** [reply]
Yeah I'm using 2.02.20. I'm running it on a raspberry pi still with debian 9. Perhaps once I get it upgraded I won't have an issue, I'll see if I can figure something out in the meantime, thanks!

**Reply by scandum** [reply]
If you can compile from source you can try editing math.c and on row 170 change:
```
/*
        sscanf(name, "%[^.]..%[^.]", strmin, strmax);

        *min = get_number(ses, strmin);
        *max = get_number(ses, strmax);
*/

        range = get_ulong(ses, name);

        *min = (int) (range / 1000000000 - 100000000);
        *max = (int) (range % 1000000000 - 100000000);
```
to
```
        sscanf(name, "%[^.]..%[^.]", strmin, strmax);

        *min = get_number(ses, strmin);
        *max = get_number(ses, strmax);
/*
        range = get_ulong(ses, name);

        *min = (int) (range / 1000000000 - 100000000);
        *max = (int) (range % 1000000000 - 100000000);
*/
```
Something about that code isn't very platform independent.

**Reply by zebulac** [reply]
That fixes my problem, thanks!

My problem is that my pi is converting the long long to double in mathexp_compute incorrectly. In the case of [2..3]:

(2 + 100000000LL) * 1000000000LL + (3 + 100000000LL) = 100000002100000003

I've verified this portion computes properly, the conversion to double yields 100000002100000000.000000.

I've been out of C programming for so long, I never have to even think about stuff like this anymore.

	#include <stdio.h>

	int main()
	{
			long double result;
			long long value = 100000002100000003LL;

			result = (long double)value;

			printf("%Lf\n", result);

			return 0;
	}

On my pi this prints 100000002100000000.000000.

**Comment by scandum** [comment]
Pretty annoying, hopefully the long double issues get straightened out eventually.

I assume you added
```
        char strmin[BUFFER_SIZE], strmax[BUFFER_SIZE];
```
to get it working.

I switched to the old code for the next release, so there should be no issues when upgrading.

**Reply by zebulac** [reply]
Yup, thanks!

I think the issue with the long to double conversion is this.

[https://stackoverflow.com/questions/33824954/cast-from-unsigned-long-long-to-double-and-vice-versa-changes-the-value](https://stackoverflow.com/questions/33824954/cast-from-unsigned-long-long-to-double-and-vice-versa-changes-the-value)

The results seem to be entirely hardware dependent.
